examples/lock-app/silabs/src/LockManager.cpp

```diff
@@ -522,23 +569,30 @@ bool LockManager::SetCredential(chip::EndpointId endpointId, uint16_t credential
                     "[credentialStatus=%u,credentialType=%u,credentialDataSize=%u,creator=%d,modifier=%d]",
                     to_underlying(credentialStatus), to_underlying(credentialType), credentialData.size(), creator, modifier);
 
-    auto & credentialInStorage = mLockCredentials[to_underlying(credentialType)][credentialIndex];
+    credentialInStorage.status             = credentialStatus;
+    credentialInStorage.credentialType     = credentialType;
+    credentialInStorage.createdBy          = creator;
+    credentialInStorage.lastModifiedBy     = modifier;
+    credentialInStorage.credentialDataSize = credentialData.size();
 
-    credentialInStorage.status         = credentialStatus;
-    credentialInStorage.credentialType = credentialType;
-    credentialInStorage.createdBy      = creator;
-    credentialInStorage.lastModifiedBy = modifier;
+    memcpy(credentialInStorage.credentialData, credentialData.data(), credentialInStorage.credentialDataSize);
```

Comment:
This is one of the reasons why I am not a fan having this as a static structure... the last added credentials will remain in RAM, which is potentially a security risk. In this case I suggest to clear the `credentialInStorage.credentialData` once we have saved it in flash but we need a better solution that ensures that we do not keep this data allocated once its owner is done using it.